pkg/collect/longhorn.go

```diff
@@ -0,0 +1,256 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "list nodes.longhorn.io")
+	}
+	dir := GetLonghornNodesDirectory(ns)
+	for _, node := range nodes.Items {
+		b, err := yaml.Marshal(node)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal node %s", node.Name)
+		}
+		key := filepath.Join(dir, node.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect volumes.longhorn.io
+	volumes, err := client.Volumes(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list volumes.longhorn.io")
+	}
+	dir = GetLonghornVolumesDirectory(ns)
+	for _, volume := range volumes.Items {
+		b, err := yaml.Marshal(volume)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal volume %s", volume.Name)
+		}
+		key := filepath.Join(dir, volume.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect replicas.longhorn.io
+	replicas, err := client.Replicas(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list replicas.longhorn.io")
+	}
+	dir = GetLonghornReplicasDirectory(ns)
+	for _, replica := range replicas.Items {
+		b, err := yaml.Marshal(replica)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal replica %s", replica.Name)
+		}
+		key := filepath.Join(dir, replica.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect engines.longhorn.io
+	engines, err := client.Engines(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list engines.longhorn.io")
+	}
+	dir = GetLonghornEnginesDirectory(ns)
+	for _, engine := range engines.Items {
+		b, err := yaml.Marshal(engine)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal engine %s", engine.Name)
+		}
+		key := filepath.Join(dir, engine.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect engineimages.longhorn.io
+	engineImages, err := client.EngineImages(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list engineimages.longhorn.io")
+	}
+	dir = GetLonghornEngineImagesDirectory(ns)
+	for _, engineImage := range engineImages.Items {
+		b, err := yaml.Marshal(engineImage)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal engineimage %s", engineImage.Name)
+		}
+		key := filepath.Join(dir, engineImage.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect instancemanagers.longhorn.io
+	instanceManagers, err := client.InstanceManagers(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list instancemanagers.longhorn.io")
+	}
+	dir = GetLonghornInstanceManagersDirectory(ns)
+	for _, instanceManager := range instanceManagers.Items {
+		b, err := yaml.Marshal(instanceManager)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to marshal instancemanager %s", instanceManager.Name)
+		}
+		key := filepath.Join(dir, instanceManager.Name+".yaml")
+		final[key] = b
+	}
+
+	// collect backingimagemanagers.longhorn.io
```

Comment:
Default kurl installs with longhorn don't have any backingimagemanagers, backingimages, or sharemanagers, but I included them for completeness.